python/cudf/cudf/core/column/decimal.py

```diff
@@ -0,0 +1,76 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.
+
+import cupy as cp
+import numpy as np
+import pyarrow as pa
+
+from cudf import _lib as libcudf
+from cudf.core.buffer import Buffer
+from cudf.core.column import ColumnBase
+from cudf.core.dtypes import DecimalDtype
+from cudf.utils.utils import pa_mask_buffer_to_mask
+
+
+class DecimalColumn(ColumnBase):
+    @classmethod
+    def from_arrow(cls, data: pa.Array):
+        mask_buf = data.buffers()[0]
+        mask = (
+            mask_buf
+            if mask_buf is None
+            else pa_mask_buffer_to_mask(mask_buf, len(data))
+        )
+        data_128 = cp.array(np.frombuffer(data.buffers()[1]).view("int64"))
+        data_64 = data_128[::2].copy()
+        return cls(
+            data=Buffer(data_64.view("uint8")),
+            size=len(data),
+            dtype=DecimalDtype.from_arrow(data.type),
```

Comment:
nit: We should probably validate the dtype before doing all of the expensive memory copies and whatnot.

Reply:
Done